rustls/examples/internal/bogo_shim.rs

```diff
@@ -610,13 +611,29 @@ fn handle_err(err: rustls::Error) -> ! {
         Error::AlertReceived(AlertDescription::InternalError) => {
             quit(":PEER_ALERT_INTERNAL_ERROR:")
         }
-        Error::CorruptMessagePayload(ContentType::Alert) => quit(":BAD_ALERT:"),
-        Error::CorruptMessagePayload(ContentType::ChangeCipherSpec) => {
+        Error::CorruptMessage(InvalidMessage::MissingPayload(ContentType::Alert)) => {
+            quit(":BAD_ALERT:")
+        }
+        Error::CorruptMessage(InvalidMessage::MissingPayload(ContentType::ChangeCipherSpec)) => {
             quit(":BAD_CHANGE_CIPHER_SPEC:")
         }
-        Error::CorruptMessagePayload(ContentType::Handshake) => quit(":BAD_HANDSHAKE_MSG:"),
-        Error::CorruptMessagePayload(ContentType::Unknown(42)) => quit(":GARBAGE:"),
-        Error::CorruptMessage => quit(":GARBAGE:"),
+        Error::CorruptMessage(InvalidMessage::MissingPayload(ContentType::Handshake)) => {
+            quit(":BAD_HANDSHAKE_MSG:")
+        }
+        Error::CorruptMessage(InvalidMessage::InvalidKeyUpdate(KeyUpdateRequest::Unknown(42))) => {
+            quit(":BAD_HANDSHAKE_MSG:")
+        }
+        Error::CorruptMessage(InvalidMessage::InvalidCertRequest)
+        | Error::CorruptMessage(InvalidMessage::InvalidDhParams)
+        | Error::CorruptMessage(InvalidMessage::MissingKeyExchange) => quit(":BAD_HANDSHAKE_MSG:"),
+        Error::CorruptMessage(InvalidMessage::InvalidContentType)
+        | Error::CorruptMessage(InvalidMessage::EmptyDataForDisallowedRecord)
+        | Error::CorruptMessage(InvalidMessage::UnknownProtocolVersion)
+        | Error::CorruptMessage(InvalidMessage::MessageTooLarge)
+        | Error::CorruptMessage(InvalidMessage::MissingPayload(ContentType::Unknown(42))) => {
+            quit(":GARBAGE:")
+        }
+        Error::CorruptMessage(InvalidMessage::IncorrectFrame) => quit(":GARBAGE:"),
```

Comment:
Can we do a catchall `Error::CorruptMessage(_)` here at some point so we don't have to exhaustively list all the cases manually?

Reply:
This is now done 👍